cmd/tetragon/main.go

```diff
@@ -577,18 +577,22 @@ func waitCRDs(config *rest.Config) error {
 }
 
 func loadTpFromDir(ctx context.Context, dir string) error {
-	tpMaxDepth := 1
-	tpFS := os.DirFS(dir)
-
-	if dir == defaults.DefaultTpDir {
-		// If the default directory does not exist then do not fail
-		// Probably tetragon not fully installed, developers testing, etc
-		if _, err := os.Stat(dir); os.IsNotExist(err) {
-			log.WithField("tracing-policy-dir", dir).Info("Loading Tracing Policies from directory ignored, directory does not exist")
-			return nil
+	if _, err := os.Stat(dir); err != nil {
+		if os.IsNotExist(err) {
+			// If the default directory does not exist then do not fail
+			// Probably tetragon not fully installed, developers testing, etc
+			if dir == defaults.DefaultTpDir {
+				log.WithField("tracing-policy-dir", dir).Info("Loading Tracing Policies from directory ignored, directory does not exist")
+				return nil
+			}
+			return fmt.Errorf("Tracing Policies directory does not exist: %s", dir)
 		}
+		return fmt.Errorf("Failed to access tracing policies directory: %w", err)
```

Comment:
```suggestion
		// Do not fail if the default directory doesn't exist,
		// it might because of developer setup or incomplete installation
		if os.IsNotExist(err) && dir == defaults.DefaultTpDir {
			log.WithField("tracing-policy-dir", dir).Info("Loading Tracing Policies from directory ignored, directory does not exist")
			return nil
		}
		return fmt.Errorf("Failed to access tracing policies dir %s: %w", dir, err)
```

Reply:
I think this relates how the code evolved to not break various tetragon versions, but also in case you also just run ./tetragon directly.

As long as we don't fail if default dir doesn't exist and we fail for all the rest is fine.

Thank you @mtardy @arthur-zhang